katsdptelstate/rdb_writer.py

```diff
@@ -97,49 +99,52 @@ def __init__(self, filename):
         self.keys_written = 0
         self.keys_failed = 0
         self._rdb_context = _rdb_file(self.filename)
-        self._fileobj = None
+        self._fileobj = self._rdb_context.__enter__()
 
     def __enter__(self):
-        """Enter RDB writer context, opening RDB file and writing header."""
-        self._fileobj = self._rdb_context.__enter__()
         return self
 
-    def __exit__(self, exc_type, exc_value, traceback):
-        """Exit RDB writer context, closing off file and deleting it if empty."""
+    def __exit__(self, *exc_info):
+        self.close()
+
+    def close(self):
+        """Close off RDB file and delete it if it contains no keys."""
         try:
-            self._rdb_context.__exit__(exc_type, exc_value, traceback)
+            self._rdb_context.__exit__(None, None, None)
         finally:
             if not self.keys_written:
                 logger.error("No valid keys found - removing empty file")
                 os.remove(self.filename)
-        return False
 
     def save(self, client, keys=None):
         """Save a specified subset of keys from a Redis DB to the RDB dump file.
 
-        This is a very limited RDB dump utility. Either a compatible redis client
-        or a :class:`katsdptelstate.Backend` must be provided. Missing or broken
+        This is a very limited RDB dump utility. It takes a Redis database
+        either from a high-level :class:`~katsdptelstate.TelescopeState` object,
+        its low-level backend or a compatible redis client. Missing or broken
         keys are skipped and logged without raising an exception.
 
         Parameters
         ----------
-        client : :class:`~redis.StrictRedis` or :class:`~katsdptelstate.telescope_state.Backend`
```

Comment:
I think the noqa has to be after the string, not inside it (you don't want it to be part of the docstring).

Reply:
Dope! Fixed.